store/sqldb/object_integrity.go

```diff
@@ -199,6 +205,78 @@ func (s *SpDBImpl) DeleteObjectIntegrity(objectID uint64, redundancyIndex int32)
 	return err
 }
 
+type ByRedundancyIndexAndObjectID []*corespdb.IntegrityMeta
+
+func (b ByRedundancyIndexAndObjectID) Len() int { return len(b) }
+
+// Less we want to sort as ascending here
+func (b ByRedundancyIndexAndObjectID) Less(i, j int) bool {
+	if b[i].ObjectID == b[j].ObjectID {
+		return b[i].RedundancyIndex < b[j].RedundancyIndex
+	}
+	return b[i].ObjectID < b[j].ObjectID
+}
+func (b ByRedundancyIndexAndObjectID) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
+
+// ListIntegrityMetaByObjectIDRange list objects info by a block number range
+func (s *SpDBImpl) ListIntegrityMetaByObjectIDRange(startObjectID int64, endObjectID int64, includePrivate bool) ([]*corespdb.IntegrityMeta, error) {
+	var (
+		totalObjects []*corespdb.IntegrityMeta
+		objects      []*corespdb.IntegrityMeta
+		err          error
+	)
+	startTime := time.Now()
+	defer func() {
+		if err != nil {
+			metrics.SPDBCounter.WithLabelValues(SPDBFailureGetObjectIntegrity).Inc()
+			metrics.SPDBTime.WithLabelValues(SPDBFailureGetObjectIntegrity).Observe(
+				time.Since(startTime).Seconds())
+		} else {
+			metrics.SPDBCounter.WithLabelValues(SPDBSuccessGetObjectIntegrity).Inc()
+			metrics.SPDBTime.WithLabelValues(SPDBSuccessGetObjectIntegrity).Observe(
+				time.Since(startTime).Seconds())
+		}
+	}()
+
+	if includePrivate {
```

Comment:
maybe we don't have to distinguish private or public objects here, as SP need gc for all zombie pieces.